### PR TITLE
bugfix: CLDSRV-47 check http header size

### PR DESCRIPTION
**Pull request by alexanderchan-scality**
return error if http headers size is too large

### CODE REVIEW

File: lib/api/apiUtils/object/checkHttpHeadersSize.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const { errors } = require('arsenal');
+const { maxHttpHeadersSize } = require('../../../../constants');
+
+/**
+ * Checks the size of the size of the HTTP headers
```
Comment (miniscruff) [inline lib/api/apiUtils/object/checkHttpHeadersSize.js]:
```suggestion
 * Checks the size of the HTTP headers
```

File: lib/api/apiUtils/object/checkHttpHeadersSize.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * @param {object} requestHeaders - HTTP request headers
+ * @return {object} object with error or null
+ */
+function checkHttpHeadersSize(requestHeaders) {
+    let httpHeadersSize = 0;
+
+    Object.keys(requestHeaders).forEach(header => {
+        httpHeadersSize += Buffer.byteLength(header, 'utf8') +
```
Comment (miniscruff) [inline lib/api/apiUtils/object/checkHttpHeadersSize.js:13]: Just want to clarify this simply takes the number of bytes for the header and value, but does not count any separators or quotes we would need if we were to send the headers as a request. Not sure if that makes any sense...

**Review by miniscruff — CHANGES_REQUESTED** [review]

File: lib/api/apiUtils/object/checkHttpHeadersSize.js (hunk 2 above)
Comment (alexanderchan-scality) [reply lib/api/apiUtils/object/checkHttpHeadersSize.js:13]: I am not sure I understand the issue being raised. Wouldn't quotes/separators be included as part of the header values (`;` or any escaped quotes)?  
I would need to check how nodejs validates the request header size

Comment (alexanderchan-scality) [reply lib/api/apiUtils/object/checkHttpHeadersSize.js:13]: (https://scality.atlassian.net/browse/S3C-5072?focusedCommentId=293869) describes the rationale behind the change

Comment (miniscruff) [reply lib/api/apiUtils/object/checkHttpHeadersSize.js:13]: So in the request you see the headers as:
`url?h1=val1&h2=val2` etc
Just want to make sure the "header size" is not the total size of that string and is instead just the key and values. That is, the equals and ampersand are ignored.

Comment (alexanderchan-scality) [reply lib/api/apiUtils/object/checkHttpHeadersSize.js:13]: I might be wrong, but I don't think the request querystring size is factored into the header size count.

**Review by miniscruff — APPROVED** [review]

**Comment by alexanderchan-scality** [comment]
ping

**Review by rahulreddy — APPROVED** [review]

**Pull request merged**